### ISSUE TITLE
docker: "docker-init": executable file not found in $PATH

### ISSUE THREAD

**Issue by brunomcuesta** [Syslifters/sysreptor#38]
Hello!
I downloaded the installation script from the repository and I needed to change all the docker compose commands to docker-compose to proceed with the installation. 

Script:
https://docs.sysreptor.com/install.sh

The error occurs at this point in the script:

```
if 
    ! docker-compose $compose_args up -d
then
    echo "Ups. Something did not work while bringing up your containers."
    exit -2
fi
```

Error message:

> docker: "docker-init": executable file not found in $PATH

It looks like it's looking for the binary docker-init in the PATH, but it doesn't find it. How can I solve the problem?

OS: Ubuntu 22.04
Docker version: 20.10.23
Docker compose version: v2.17.2

Thanks!

**Comment by aronmolnar** [comment]
Hi,
It looks like you have docker compose v1 installed, which is deprecated and will be removed in a few weeks (see https://docs.docker.com/compose/).

Can you try to clean up your docker installation and reinstall following the official guide at https://docs.docker.com/engine/install/ubuntu/

This will also install docker-compose-plugin via the apt-get install command.

(Then you will have to use `docker compose` instead of `docker-compose` again.)

**Comment by brunomcuesta** [comment]
Docker has been updated, but it keeps showing the same error.

```
[+] Running 2/3
 ✔ Network sysreptor_default  Created                                                                                                                                                                       0.1s
 ✔ Container sysreptor-db     Healthy                                                                                                                                                                       3.2s
 ⠿ Container sysreptor-app    Starting                                                                                                                                                                      3.3s
Error response from daemon: exec: "docker-init": executable file not found in $PATH
Ups. Something did not work while bringing up your containers.

```
Docker version 24.0.1, build 6802122
Docker Compose version v2.17.2

Does docker-compose also need to be updated?

**Comment by aronmolnar** [comment]
Have you followed the steps in the official docker installation guide?

`sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin`

**Comment by brunomcuesta** [comment]
Hello!
Yes, I checked and all packages were installed. Could it be something related to the configuration?

**Comment by aronmolnar** [comment]
Can you run `docker run hello-world`?

And how about `curl -s https://docs.sysreptor.com/install.sh | bash`?

**Comment by brunomcuesta** [comment]
Hello!
Hello-world was executed successfully. It didn't show any errors with docker.

```
sudo docker run hello-world

Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
719385e32844: Pull complete
Digest: sha256:fc6cf906cbfa013e80938cdf0bb199fbdbb86d6e3e013783e5a766f50f5dbce0
Status: Downloaded newer image for hello-world:latest

Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (amd64)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
$ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker ID:
https://hub.docker.com/

For more examples and ideas, visit:
https://docs.docker.com/get-started/


docker images hello-world

REPOSITORY    TAG       IMAGE ID       CREATED       SIZE
hello-world   latest    9c7a54a9a43c   3 weeks ago   13.3kB

docker ps -a

CONTAINER ID   IMAGE       COMMAND  CREATED         STATUS                     PORTS NAMES
fceecee20d74   hello-world "/hello" 4 minutes ago   Exited (0) 4 minutes ago   xenodochial_curie
```

**Comment by MWedl** [comment]
I think the problem is related to the init system which is used for some containers in `docker-compose.yml`. The init system binary seems to be missing on your system. On my system (OS: Fedora 36, Docker version: 24.0.2) the `docker-init` binary is located at `/usr/libexec/docker/docker-init`.

Can you try to run `docker run --init hello-world`.

**Comment by brunomcuesta** [comment]
Hello!
Docker was installed following the documentation on the official website.

**Comment by aronmolnar** [comment]
Do you have the option to start off with a clean Ubuntu installation?

**Comment by aronmolnar** [comment]
Can you run `sudo snap remove docker --purge` and reboot?

**Comment by brunomcuesta** [comment]
Solved!
I removed the docker that was running in the snap. I checked the dockerd pids and removed them with the kill command. Then I started docker with the systemctl command.
These two tutorials also helped resolve the "cannot connect to the Docker daemon" error, after removing the docker that was running in snap.
https://docs.docker.com/config/daemon/troubleshoot/
https://phoenixnap.com/kb/cannot-connect-to-the-docker-daemon-error
Thank you for your help!
